### PR TITLE
IGV renderer

### PR DESCRIPTION
**Pull request by fiskus**
# Description

Added renderer of JSON using igv.js. URLs found in JSON are processed and signed

Test package is `/b/fiskus-sandbox-dev/packages/fiskus/igv/` on staging

# TODO

- [ ] ~Unit tests~
- [ ] ~Automated tests (e.g. Preflight)~
- [x] Documentation
    - [ ] ~[Python: Run `build.py`](../gendocs/build.py) for new docstrings~
    - [x] JavaScript: basic explanation and screenshot of new features
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

File: docs/Catalog/VisualizationDashboards.md [hunk 1]
```diff
@@ -72,6 +73,7 @@ or an object with one or more of the following properties:
 - `types` - a list of rendering types; currently only singleton list values are supported:
```
Comment (akarve) [inline docs/Catalog/VisualizationDashboards.md:73]:
```suggestion
- `types` - a list of rendering types (at present only singleton lists are supported):
```

File: docs/Catalog/VisualizationDashboards.md [hunk 2]
```diff
@@ -350,3 +352,22 @@ All filters and columns will be restored:
...
+
+```json
+// quilt_summarize.json
+[
+  {
+    "path": "igv-options-file.json",
+    "title": "Awesome genome",
+    "types": ["igv"]
+  }
+]
```
Comment (akarve) [inline docs/Catalog/VisualizationDashboards.md:372]:
```suggestion

## Integrative Genomics Viewer (IGV)


```suggestion
To render genome tracks, you can select "View as IGV" in the catalog,
or you can invoke [igv.js](https://igv.org/) with quilt_summarize, as shown below:
```json
// quilt_summarize.json
[
  {
    "path": "igv-options-file.json",
    "title": "Awesome genome",
    "types": ["igv"]
  }
]
```
In the above example, `igv-options-file.json` is an
[IGV browser configuration](https://github.com/igvteam/igv.js/wiki/Browser-Creation#browser-configuration-options).
```

File: shared/schemas/quilt_summarize.json [hunk 3]
```diff
@@ -44,7 +44,7 @@
 
     "typeShorthand": {
       "type": "string",
-      "enum": ["echarts", "json", "jupyter", "perspective", "vega", "voila"]
+      "enum": ["echarts", "igv", "json", "jupyter", "perspective", "vega", "voila"]
```
Comment (akarve) [inline shared/schemas/quilt_summarize.json:47]: Isn't `"IGV"` more correct? It's definitely what we should show in the dropdown.

File: docs/Catalog/VisualizationDashboards.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ exploration:
 * [ECharts](#echarts)
 * [Voila](#voila) (Developer preview)
 * [Perspective](#perspective)
+* [Igv](#igv)
```
Comment (akarve) [inline docs/Catalog/VisualizationDashboards.md:20]:
```suggestion
* [IGV](#igv)
```

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,7 @@
 * [Changed] Disable upload optimization for objects with SSE-KMS ([#2790](https://github.com/quiltdata/quilt/pull/2790))
 
 ## Catalog, Lambdas
+* [Added] Add Igv renderer ([#2965](https://github.com/quiltdata/quilt/pull/2965))
```
Comment (akarve) [inline docs/CHANGELOG.md:14]:
```suggestion
* [Added] Add IGV renderer ([#2965](https://github.com/quiltdata/quilt/pull/2965))
```

File: docs/Catalog/VisualizationDashboards.md [hunk 6]
```diff
@@ -350,3 +352,22 @@ All filters and columns will be restored:
...
 ]
 ```
+
+## Igv
+
+To render an [Igv](https://igv.org/), you provide a JSON file (a dictionary that
+specifies the Igv [browser configuration options](https://github.com/igvteam/igv.js/wiki/Browser-Creation#browser-configuration-options))
+and you set the `"types"` property to `[ "igv" ]`.
+
+Alternatively, you can browse that JSON file in Quilt web and switch viewer from JSON to Igv
```
Comment (akarve) [inline docs/Catalog/VisualizationDashboards.md:362]:
```suggestion
## Integrative Genomics Viewer (IGV)

To render genome tracks, you can select "View as IGV" in the catalog,
or you can invoke [igv.js](https://igv.org/) in quilt_summarize, as shown below:
```

File: docs/Catalog/VisualizationDashboards.md [hunk 7]
```diff
@@ -350,3 +352,22 @@ All filters and columns will be restored:
...
+```json
+// quilt_summarize.json
+[
+  {
+    "path": "igv-options-file.json",
+    "title": "Awesome genome",
+    "types": ["igv"]
+  }
+]
+```
```
Comment (akarve) [inline docs/Catalog/VisualizationDashboards.md:373]: It's impossible to make a suggestion with three backticks in GH. Anyway, please add this sentence on the next line:
```
In the above example, `igv-options-file.json` is an
[IGV browser configuration](https://github.com/igvteam/igv.js/wiki/Browser-Creation#browser-configuration-options).
```

**Review by akarve — APPROVED** [review]

File: shared/schemas/quilt_summarize.json (hunk 3 above)
Comment (fiskus) [reply shared/schemas/quilt_summarize.json:47]: Another part of the code is responsible for titles in the dropdown. I fixed "Igv" -> "IGV" there

**Review by nl0 — APPROVED** [review]
this one is quite large, so i can't say i understand everything happening here, but it looks ok as far as i can tell

**Comment by fiskus** [comment]
Fixed z-index (some elements of IGV were above "Expand" button), and re-tested

**Review by nl0 — APPROVED** [review]

**Pull request merged**